src/main/java/com/amazon/inspector/jenkins/amazoninspectorbuildstep/sbomgen/SbomgenDownloader.java

```diff
@@ -1,41 +1,46 @@
 package com.amazon.inspector.jenkins.amazoninspectorbuildstep.sbomgen;
-
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import hudson.FilePath;
-
-import javax.management.openmbean.InvalidKeyException;
 import java.io.BufferedInputStream;
 import java.io.IOException;
 import java.net.URL;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
 
 public class SbomgenDownloader {
+
+    private static final String BASE_URL = "https://amazon-inspector-sbomgen.s3.amazonaws.com/latest/linux/%s/inspector-sbomgen.zip";
+
     public static String getBinary(String configInput, FilePath workspace) throws IOException, InterruptedException, ExecutionException {
         String url = getUrl(configInput);
         FilePath zipPath = downloadFile(url, workspace);
         return unzipFile(zipPath);
     }
 
     private static String getUrl(String configInput) {
-        final String linuxAmd64Url = "https://amazon-inspector-sbomgen.s3.amazonaws.com/latest/linux/amd64/inspector-sbomgen.zip";
-        final String linuxArm64Url = "https://amazon-inspector-sbomgen.s3.amazonaws.com/latest/linux/arm64/inspector-sbomgen.zip";
-
-        switch (configInput) {
-            case "linuxAmd64":
-                return linuxAmd64Url;
-            case "linuxArm64":
-                return linuxArm64Url;
-        }
-        if ("linuxAmd64Url".equals(configInput)) {
-            return linuxAmd64Url;
-        }
-        if ("linuxArm64Url".equals(configInput)) {
-            return linuxArm64Url;
+        String osName = System.getProperty("os.name").toLowerCase();
+        if (!osName.contains("linux")) {
+            throw new UnsupportedOperationException("Unsupported OS: " + osName);
         }
 
+        String architecture = "amd64";
 
-        throw new InvalidKeyException("No url corresponding to " + configInput);
+        String osArch = System.getProperty("os.arch").toLowerCase();
```

Comment:
Log osArch here.

Reply:
System.out.println won't print to the output, it writes to a log file. Use logger.println instead.